src/gui.py

```diff
@@ -622,21 +576,587 @@ def get_data_processor() -> DataProcessor:
+                    self.validate_time_range_end()
+
+            def on_focus_out_validate_time_range_end(self, _):
+                self.validate_time_range_end()
+
+            def on_key_press_validate_data_time_step(self, event):
+                if event.keysym == "Return":
+                    self.validate_data_time_step()
+
+            def on_focus_out_validate_data_time_step(self, _):
+                self.validate_data_time_step()
+
+            def on_key_press_validate_cells_side_count_latitude(self, event):
+                if event.keysym == "Return":
+                    self.validate_cells_side_count_latitude()
+
+            def on_focus_out_validate_cells_side_count_latitude(self, _):
+                self.validate_cells_side_count_latitude()
+
+            def on_key_press_validate_cells_side_count_longitude(self, event):
+                if event.keysym == "Return":
+                    self.validate_cells_side_count_longitude()
+
+            def on_focus_out_validate_cells_side_count_longitude(self, _):
+                self.validate_cells_side_count_longitude()
+
+            def on_focus_out_validate_point_side_size(self, event):
+                if event.keysym == "Return":
+                    self.validate_point_side_size()
+
+            def on_key_press_validate_point_side_size(self, _):
+                self.validate_point_side_size()
+
+            def validate_coordinates_top(self):
+                value = self.top_coord_input.get()
+                if value:
+                    if -90 <= float(value) <= 90 and float(value) > self.down_coord:
+                        self.top_coord = float(value)
+                        self.top_coord_validation_label.config(text="Valid value")
+                        self.correctly_set_parameters[0] = 1
+                        self.check_all_parameters_validity_and_refresh_image()
+                        return True
+                    self.top_coord_validation_label.config(text="Invalid value")
+                    self.confirm_and_continue.config(state=DISABLED)
+                    self.correctly_set_parameters[0] = 0
+                return False
+
+            def validate_coordinates_down(self):
+                value = self.down_coord_input.get()
+                if value:
+                    if -90 <= float(value) <= 90 and float(value) < self.top_coord:
+                        self.down_coord = float(value)
+                        self.down_coord_validation_label.config(text="Valid value")
+                        self.correctly_set_parameters[1] = 1
+                        self.check_all_parameters_validity_and_refresh_image()
+                        return True
+                    self.down_coord_validation_label.config(text="Invalid value")
+                    self.confirm_and_continue.config(state=DISABLED)
+                    self.correctly_set_parameters[1] = 0
+                return False
+
+            def validate_coordinates_left(self):
```

Comment:
DRY!!!!!!!!!!!! - there is a lot of copy-paste code
maybe not in that PR, but we should think about cleaning that